README.md

```diff
@@ -0,0 +1,526 @@
+-->
+
+## Overview
+
+S7 Micro Service - device service for connecting Siemens S7(S7-200, S7-300, S7-400, S7-1200, S7-1500) devices by `ISO-on-TCP` to EdgeX.
+
+## Features
+
+- Single Read and Write
+- Multiple Read and Write
+- High performance, more 2000 items per second(depends on S7 model)
+  - Use `S7-Device01` sample device configuration, `interval` should less than `IdelTimeout`
+  - S7-1200 and S7-1500 preferred
+  - Create multiple connections to one S7 device use different device name
+
+## Prerequisites
+
+- A Siemens S7 series device with network interface
+- Enable ISO-on-TCP connection
+
+### Install and Deploy
+
+Clone and Build
+
+```shell
+git clone https://github.com/edgexfoundy-holding/device-s7.git
+cd device-s7
+make build
+```
+
+To start the device service:
+
+```shell
+export EDGEX_SECURITY_SECRET_STORE=false
+make run
+```
+
+To rebuild after making changes to source:
+
+```shell
+make clean
+make build
+```
+
+## Packaging
+
+This component is packaged as Docker image.
+
+For docker, please refer to the `Dockerfile`.
+
+### Build Docker image
+
+```shell
+make docker
+```
+
+The docker image looks like:
+
+```
+edgexfoundry/device-s7    3.1.0-dev    c84a77f45860   5 minutes ago   33.6MB
+```
+
+### Docker compose file
+
+Add to your docker-compose.yml.
+
+```yaml
+device-s7:
+  container_name: edgex-device-s7
+  depends_on:
+    consul:
+      condition: service_started
+      required: true
+    core-data:
+      condition: service_started
+      required: true
+    core-metadata:
+      condition: service_started
+      required: true
+  environment:
+    EDGEX_SECURITY_SECRET_STORE: 'false'
+    SERVICE_HOST: edgex-device-s7
+  hostname: edgex-device-s7
+  image: edgexfoundry/device-s7:0.0.0-dev
+  networks:
+    edgex-network: null
+  ports:
+    - mode: ingress
+      host_ip: 127.0.0.1
+      target: 59994
+      published: '59994'
+      protocol: tcp
+  read_only: true
+  restart: always
+  security_opt:
+    - no-new-privileges:true
+  user: 2002:2001
+  volumes:
+    - type: bind
+      source: /etc/localtime
+      target: /etc/localtime
+      read_only: true
+      bind:
+        create_host_path: true
+```
+
+## Usage
```

Comment:
Note that once approved and adopted into EdgeX, these sections below need to be moved into edgex-docs . Each device service has the following doc sections:
- Purpose
- Getting Started
- How-to_Guides (optional)
- Configuration
- Protocol Properties
- API Reference
- Source Code

See Device REST docs here as a guide/sample
https://docs.edgexfoundry.org/3.1/microservices/device/services/device-rest/Purpose/

Please rework below section so they fit into the above sections below and add the above section that are not coved by what you have below. This will make it easy to move into edgex-docs later.

Reply:
I'll write the `Purpose` on edgex-docs, then create a pr.